### ISSUE TITLE
Loss of privacy (scope) in communicating with Hubzilla (8.41)

### ISSUE THREAD

**Issue by jayrope** [mastodon/mastodon#25061]
### Steps to reproduce the problem

1. Sent a direct message from mastodon.online to a channel on Hubzilla 8.41
2. Channel owner respondes with a DM
3. When answering back from Mastodon just pressing reply changes the scope of the conversation from direct message to "followers only", making subsequent messages of a private conversation visible to followers of the Mastodon profile.

We've recently quickly resolved another interoperability issue (correct display of tags in Mastodon which were sent from Hubzilla). 
If anyone should work on this issue now -> i'd like to encourage you to report your findings on 
https://framagit.org/hubzilla/core/-/issues as well, as this had helped communication with Hubzilla devs a lot before. 

**Thank you very much in advance!**

### Expected behaviour

Keep conversation privacy scope 

### Actual behaviour

Subsequent change of privacy scope upon direct message from Hubzilla 

### Detailed description

Subsequent privacy scope upon direct message from Hubzilla was changed without visual notification. 
I only noticed, because i was double checking on scope, after i had noticd, that my Mastodon answer to the DM from Hubzilla was send to all my followers. 

### Specifications

Mastodon 4.12 - namely the version mastoon.online uses atm.
Hubzila 8.41 as conversation partner.

**Comment by trwnh** [comment]
not sure, but maybe it's possible the status is not actually "followers", but "limited"? this usually happens when there is someone addressed in `to` or `cc` that was not mentioned in `tag`. unfortunately i cannot comment further without knowing what the payload of the hubzilla reply was, and if i understand you correctly, the reply was a private mention.

**Comment by jayrope** [comment]
@trwnh In Mastodon we appear to be both following each other.

Now i am seeing, that despite DM from Hubzilla appearing _within_ a DM thread, it is not marked "private mention" itself (and has a white background instead of a lightly blue), 
while my own messages are marked like this. 

So, do i understand you right: We do need an example thread for this situation?

**Comment by trwnh** [comment]
yes, scope is not per-thread but per-post. likely hubzilla is adding a recipient here without mentioning them, causing the "direct" status to become a "limited" status, which is only half supported in mastodon. they are stored in the database as "limited" but serialized in the API as "private" (followers only). attempting to reply in your mastodon client/app is probably only allowing you to author your reply as "private" as i don't think you can actually author "limited" statuses via the API yet?

**Comment by ClearlyClaire** [comment]
On Mastodon's side, do you get a notification when receiving the Hubzilla DM that shows as followers-only on Mastodon?

**Comment by jayrope** [comment]
Hello ClearlyClaire 

> On Mastodon's side, do you get a notification when receiving the Hubzilla DM that shows as followers-only on Mastodon?

Yes. In fact i get two of the same notifications and one post in the @ private mentions. 
Before i only got one notification.

Also, now i see **no privacy scope change**. 
When pressing reply in Mastodon DM scope is preserved. 

Did you already change anything?

**Comment by ClearlyClaire** [comment]
> **NOW when answering back from HZ, privacy scope is NOT preserved anymore: Mastodon resets to "followers_only"**

I got that. Do you get a notification for *this* message in Mastodon?

**Comment by trwnh** [comment]
@git-marijus i think collections/lists are stripped from recipients because there's no way to unpack them without knowing their contents. unpacking should generally be done at the delivery side, not the recipient side. the one exception that mastodon makes is if it's a `followers` property on any known account's associated actor

**Comment by git-marijus** [comment]
@trwnh not sure if it's a good idea to give people the impression they received a private mention while they did not. I would consider this a security issue. I think it is legitimate to list collections in either TO or CC.

**Comment by nemobis** [comment]
https://github.com/mastodon/mastodon/issues/24523 is related
